src/binder/statement/create_table.rs

```diff
@@ -45,73 +45,121 @@ impl Binder {
-                    }
-                    col.set_id(idx as ColumnId);
-                    col_catalogs.push(col);
+                if ordered_pk_ids.len() > 1 {
+                    // multi primary key should be declared by "primary key(c1, c2...)" syntax
+                    return Err(BindError::NotSupportedTSQL);
+                }
+
+                let pks_name_from_constraints = Binder::pks_name_from_constraints(constraints);
+                if has_pk_from_column && !pks_name_from_constraints.is_empty() {
+                    // can't get primary key both from "primary key(c1, c2...)" syntax and
+                    // column's option
+                    return Err(BindError::NotSupportedTSQL);
+                } else if !has_pk_from_column {
+                    ordered_pk_ids =
+                        Binder::ordered_pks_from_constraint(&pks_name_from_constraints, columns);
+                }
+
+                let mut columns: Vec<ColumnCatalog> = columns
+                    .iter()
+                    .enumerate()
+                    .map(|(idx, col)| {
+                        let mut col = ColumnCatalog::from(col);
+                        col.set_id(idx as ColumnId);
+                        col
+                    })
+                    .collect();
+
+                // // TODO: when remove `is_primary` filed in `ColumnDesc`,
+                // // Remove this line and change `columns` above to immut.
+                for &index in &ordered_pk_ids {
+                    columns[index as usize].set_primary(true);
                 }
 
                 Ok(BoundCreateTable {
                     database_id: db.id(),
                     schema_id: schema.id(),
                     table_name: table_name.into(),
-                    columns: col_catalogs,
+                    columns,
                     ordered_pk_ids,
                 })
             }
             _ => panic!("mismatched statement type"),
         }
     }
 
-    fn get_extra_pks(constraints: &Vec<TableConstraint>) -> HashMap<String, usize> {
-        let mut ordered_pk_pos: HashMap<String, usize> = HashMap::new();
+    // get primary keys' id in declared order。
+    // we use index in columns vector as column id
+    fn ordered_pks_from_columns(columns: &[ColumnDef]) -> Vec<ColumnId> {
+        let mut ordered_pks = Vec::new();
+
+        for (index, col_def) in columns.iter().enumerate() {
+            for option_def in &col_def.options {
+                let is_primary_ = if let ColumnOption::Unique { is_primary } = option_def.option {
+                    is_primary
+                } else {
+                    false
+                };
+                if is_primary_ {
+                    ordered_pks.push(index as ColumnId);
+                }
+            }
+        }
+        ordered_pks
+    }
+
+    // We have used `pks_name_from_constraints` to get the primary keys' name sorted by declaration
```

Comment:
Use `///` for comments on functions, so that Rust will recognize it as docstring, and you can view it in `make docs`.